examples/seq2seq/run_eval_search.py

```diff
@@ -0,0 +1,96 @@
+import argparse
+import itertools
+import sys
+
+from run_eval import run_generate
+
+
+def parse_search_arg(search):
+    groups = search.split()
+    entries = {k: vs for k, vs in (g.split("=") for g in groups)}
+    entry_names = list(entries.keys())
+    sets = [list((f"--{k} {v}") for v in vs.split(":")) for k, vs in entries.items()]
+    matrix = [list(x) for x in itertools.product(*sets)]
+    return matrix, entry_names
+
+
+def run_search():
+    """
+     Run parametric search over the desired hparam space with help of ``run_eval.py``.
+
+     All the arguments except ``--search`` are passed to ``run_eval.py`` as is. The values inside of "--search" are parsed, reformatted and fed to ``run_eval.py`` as additional args.
+
+    The format for the ``--search`` value is a simple string with hparams and colon separated values to try, e.g.:
+    ```
+     --search "num_beams=5:10 length_penalty=0.8:1.0:1.2 early_stopping=true:false"
+    ```
+    which will generate ``12`` ``(2*3*2)`` searches for a product of each hparam. For example the example that was just used will invoke ``run_eval.py`` repeatedly with:
+
+    ```
+     --num_beams 5 --length_penalty 0.8 --early_stopping true
+     --num_beams 5 --length_penalty 0.8 --early_stopping false
+     [...]
+     --num_beams 10 --length_penalty 1.2 --early_stopping false
+    ```
+
+    On completion, this function prints a markdown table of the results sorted by the best BLEU score and the winning arguments.
+
+
+    """
+    prog = sys.argv[0]
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--search",
+        type=str,
+        required=False,
+        help='param space to search, e.g. "num_beams=5:10 length_penalty=0.8:1.0:1.2"',
+    )
+
+    args, args_main = parser.parse_known_args()
+    args_normal = [prog] + args_main
+
+    matrix, col_names = parse_search_arg(args.search)
+    col_names.insert(0, "bleu")
```

Comment:
This assumes `--task=translation`. 
There are multiple metrics returned for summarization.

Reply:
+ Would be 🔥  (if easy) to factor this out somehow so that easier to use w distributed.

Reply:
ah, good point. Perhaps let's start with supporting just translation? I will check that `task==translation` and bail otherwise? Then we can extend to others.

Reply:
I haven't worked with others yet, so I won't know what to look for.

Reply:
or if it's transparent, can you give me a table of `task | scorename` and then I can use it to code the right look up.

Reply:
wrt, distributed - we will sort it out once it's in place

Reply:
`calculate_rouge` returns `{'rouge1': 45.1224, 'rouge2': 22.9417, 'rougeL': 37.7742,}`
So it would need 3 columns :(

Reply:
OK, solved this one, currently have:
```
# A table of supported tasks and the list of scores in the order of importance to be sorted by.
# To add a new task, simply list the score names that `run_eval.run_generate()` returns
task_score_names = {
    "translation": ["bleu"],
    "summarization": ["rouge1", "rouge2", "rougeL"],
}
```
on top. 

It should automatically pick up the right columns and sort them in that order.  Only need to edit this map.

Please let me know what other tasks and scores to add.